tests/unit/clients/test_http.py

```diff
@@ -212,3 +212,68 @@ def simple_json_callback(resp):
     assert exp_resp == resp
     assert 1 == mock_set_valid_token_called
     assert 2 == len(caplog.records)
+
+
+@pytest.mark.parametrize('post_arg', ('json', 'body'))
+@pytest.mark.asyncio
+async def test_post_data(post_arg, client, monkeypatch, caplog):
+    """HTTP POST request."""
+    mock_set_valid_token_called = 0
+
+    async def mock_set_valid_token():
+        nonlocal mock_set_valid_token_called
+        mock_set_valid_token_called += 1
+
+    monkeypatch.setattr(client, 'set_valid_token', mock_set_valid_token)
+
+    post_data = {'hello': 'world'}
+
+    request_kwargs = {
+        'method': 'post',
+        'url': conftest.API_URL,
+        post_arg: post_data
+    }
+
+    with aioresponses() as mocked:
```

Comment:
I probably just need to do some reading, but how does this work?

Reply:
aioresponses is a nifty library to make it easier to mock http interactions. Instead of having to manually mock and patch things on aiohttp, aioreponses does all that for you. You have to specify the endpoint URL you expect to hit and its response first like the line below. If you were mocking out a get request, it'd look like:

```python
mocked.get('http://example.com', body='<h1>Hello world</h1>')
```

Lemme know if you wanna talk more about it!

Link: https://github.com/pnuckowski/aioresponses